README.md

```diff
@@ -69,29 +69,31 @@ Only the `source` import source phase is specified by this proposal.
 
 ### Defining Module Source
 
-Module source is defined through the [ECMA-262 ES modules HostLoadImportedModule refactoring][],
-which moves the construction of module records to the host, and on which we can define a
-`[[ModuleSourceObject]]` custom object to be returned by module source phase that is compatible
-with module expressions.
+The object provided by the module source phase must have a prototype set to the
+`ModuleSourcePrototype` object, defined by this specification to be a minimal
+object representation for a compiled modular resource.
 
 ### JS Module Source
 
-The type of the module source object for a JavaScript module is would depend
-on the [module source][] specification.
+For JavaScript modules, the module source phase is then specified to return
+a `ModuleSource` object, representing an ECMAScript Module Source.
 
-In the current specification, the JS module source case just throws an error
-to support this as a future addition.
+This is a minimal implementation to support the source phase including branding
+checks via `toStringTag` and providing a unique key for the module resource.
+
+Future proposals may then add support for [bindings lookup methods][],
+the [ModuleSource constructor][] and [instantiation][] support.
+
+New properties may be added to the base `ModuleSourcePrototype`, or shared
+with ECMASCript module sources via `ModuleSource` additions.
 
 ### Wasm Module Source
 
-The type of the module source object for WebAssembly would be a
-`WebAssembly.Module`, as defined in the [WebAssembly JS integration
-API][wasm-js-api].
+For WebAssembly modules, the existing `WebAssembly.Module` object is to be
+updated to be a subclass of `AbstractModuleSource` in the
```

Comment:
I don’t see `AbstractModuleSource` here. Should there be a reference? I’m looking for the object graph you propose so one can walk by property from `globalThis` to `AbstractModuleSource`. I assume directly and that `ModuleSource` is a subclass, or prefer entirely for æsthetic reasons that I hold lightly, the omission of `AbstractModuleSource` in favor of `globalThis.ModuleSource` serving as both the base class for all language module sources and also the JavaScript module source. I will be satisfied for purposes of review by either detail.

Reply:
I originally had an explicit `AbstractModuleSource`, but then saw we have existing precedent for prototype objects which are not themselves constructors (intrinsics ending in `Prototype`), so thought we could try that model instead.

So the chain is now effectively:

```js
global.ModuleSource = function ModuleSource () {
  throw new Error('unimplemented');
}
ModuleSource.prototype = ModuleSourcePrototype;
```

where ModuleSourcePrototype is then just defined as an object.

Reply:
That’s fine too, though prototypes usually participate in a `constructor.prototype` property cycle.